src/android/code-tools.ts

```diff
@@ -0,0 +1,157 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as Sentry from '@sentry/node';
+// @ts-ignore - clack is ESM and TS complains about that. It works though
+import * as clack from '@clack/prompts';
+import chalk from 'chalk';
+import {
+  sentryImport,
+  sentryImportKt,
+  testErrorSnippet,
+  testErrorSnippetKt,
+} from './templates';
+
+/**
+ * Looks in src/main/java or src/main/kotlin for the specified {@link packageName} and
+ * {@link activityName} by concatenating them. For example:
+ *
+ * src/
+ *   main/
+ *     java/ or kotlin/
+ *       my.package.name/
+ *         ui/
+ *           MainActivity.kt
+ *
+ * src/main/java can contain both .java and .kt sources, whilst src/main/kotlin only .kt
+ *
+ * @param appDir
+ * @param packageName
+ * @param activityName
+ * @returns path to the Main Activity
+ */
+export function findActivitySourceFile(
+  appDir: string,
+  packageName: string,
+  activityName: string,
+): string | undefined {
+  const javaSrcDir = path.join(appDir, 'src', 'main', 'java');
+  let possibleActivityPath;
+  const packageNameParts = packageName.split('.');
+  const activityNameParts = activityName.split('.');
+
+  if (fs.existsSync(javaSrcDir)) {
+    possibleActivityPath = `${path.join(
+      javaSrcDir,
+      ...packageNameParts,
+      ...activityNameParts,
+    )}.java`;
+    if (!fs.existsSync(possibleActivityPath)) {
+      // try kotlin if no java file
+      possibleActivityPath = possibleActivityPath.replace('.java', '.kt');
```

Comment:
l: Here we want to look up a file that could either end with `.java` or `.kt`, right? There's a utility function that does this for various JS file endings. Maybe we can refactor it to be more general and reuse it here: https://github.com/getsentry/sentry-wizard/blob/1d62aab0b4cedbfe84b1f20126831efae09568f2/src/utils/ast-utils.ts#L5-L14

Feel free to ignore, logaf-l, I just think it might simplify this function a little.